code/mob/living/stamina_stuns.dm

```diff
@@ -91,6 +91,36 @@
 	if(src.stamina_bar?.last_update != TIME) src.stamina_bar?.update_value(src)
 	return
 
+/// Adds 'get up' stamina/stun reduction, from taking beatdown damage.
+/mob/proc/get_revenge_stamina(stamina_damage, brute, burn, damage_type )
+	return
+
+/mob/living/get_revenge_stamina(stamina_damage, brute, burn, damage_type )
+	. = ..()
+	if(!src.use_stamina) return
+	if(!isnum(stamina_damage)) return
+	if (src.hasStatus("weakened") && !src.hasStatus("paralysis") && (brute > 0 || burn >= 5))
+
+		var/weak_duration = src.getStatusDuration("weakened")
+		if (weak_duration > 3 SECONDS) //if we have a big stun, we can kick it down a lot
+			var/stun_reduction = min(weak_duration-3 SECONDS,(brute + burn)*0.6 SECONDS) // let's saaay 1 full stun is 50 health.
+			var/stunres_penalty = clamp(1-(get_stun_resist_mod()/2)/100,0,1)
+			src.setStatus("weakened", weak_duration-max(1,stun_reduction*stunres_penalty))
+		else if (weak_duration > 0 SECONDS)  // but we also still need a penalty for getting stamcrit which is 5s.
+			src.setStatus("weakened", weak_duration-1)
+
+		var/stun_duration = src.getStatusDuration("stunned")
+		if (stun_duration > 3 SECONDS)
+			var/stun_reduction = min(stun_duration-3 SECONDS,(brute + burn)*0.6 SECONDS)
+			var/stunres_penalty = clamp(1-(get_stun_resist_mod()/2)/100,0,1)
+			src.setStatus("weakened", stun_duration-max(1,stun_reduction*stunres_penalty))
+		else if (stun_duration > 0 SECONDS)
+			src.setStatus("weakened", stun_duration-1)
```

Comment:
```suggestion
```
I've gone ahead and removed the unneeded "stunned" portion of projectile stuns so this can be removed (it also wouldn't have worked due to bad copy paste)

Reply:
ick. my bad